### PR TITLE
New filters for programs and clips

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: src/components/Sections/Topics/SubTopics.js [hunk 1]
```diff
@@ -43,6 +43,8 @@ const SubTopics = ({ namespace, rootID, t }) => {
   const root  = getTagById(rootID);
   const items = useMemo(() => getItemsRecursive(rootID, getTagById, baseItems) || [], [rootID, getTagById, baseItems]);
 
+  useEffect(() => setQuery(null), []);
```
Comment (bbfsdev) [inline src/components/Sections/Topics/SubTopics.js:46]: What setQuery does?

File: src/components/Sections/Topics/TopicContainer.js [hunk 2]
```diff
@@ -107,7 +107,7 @@ const TopicContainer = ({ t }) => {
   const dispatch = useDispatch();
 
   useEffect(() => {
-    dispatch(actions.fetchStats(namespace, {}, true));
+    dispatch(actions.fetchStats(namespace, {}, { isPrepare: true, countC: true, countL: true }));
```
Comment (bbfsdev) [inline src/components/Sections/Topics/TopicContainer.js:110]: What is this {} action? What is it for?

File: src/components/shared/ContentItem/TextListTemplate.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ const TextListTemplate = ({ label, unit, t }) => {
       key={id}
       className="text_item"
     >
-      <Image verticalAlign="top">
+      <Image verticalAlign="top" >
```
Comment (bbfsdev) [inline src/components/shared/ContentItem/TextListTemplate.js:27]: Remove redundant space after "top"

File: src/components/shared/TooltipIfNeed.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import { useSelector } from 'react-redux';
+import { Popup, Ref } from 'semantic-ui-react';
+
+import { selectors as settings } from '../../redux/modules/settings';
+import { isLanguageRtl } from '../../helpers/i18n-utils';
+
+/**
+ *
+ * @param props
+ * text: text that will be on the tooltip.
```
Comment (bbfsdev) [inline src/components/shared/TooltipIfNeed.js:11]: Should we add such description for each hook we write?

File: src/sagas/stats.js [hunk 5]
```diff
@@ -43,7 +43,7 @@ export function* callUnitsStats(args, namespace) {
 }
 
 function* watchFetchList() {
-  yield takeLatest([lists.FETCH_LIST, tags.FETCH_STATS], fetchCUStats);
+  yield takeLatest([tags.FETCH_STATS], fetchCUStats);
```
Comment (bbfsdev) [inline src/sagas/stats.js:46]: Why FETCH_LIST is not relevant any more.

File: src/sagas/recommended.js [hunk 6]
```diff
@@ -268,29 +288,6 @@ function* fetchWatchingNow(uids) {
   }
 }
 
-function* fetchMissingUnits(recommendedItems) {
```
Comment (bbfsdev) [inline src/sagas/recommended.js:271]: Why did you move this?

File: src/sagas/mdb.js [hunk 7]
```diff
@@ -132,13 +130,35 @@ export function* fetchLabels(action) {
...
     const { data } = yield call(Api.labels, params);
-    yield put(mdb.receiveLabels(data.labels));
+    yield put(actions.receiveLabels(data.labels));
     yield put(actions.fetchLabelsSuccess(data));
   } catch (err) {
     console.error('fetchLabels errors', err);
   }
 }
 
+export function* fetchMissingUnits(uids) {
```
Comment (bbfsdev) [inline src/sagas/mdb.js:140]: Where do you use this in this file?
Why did you add this to this file?

File: src/sagas/mdb.js [hunk 8]
```diff
@@ -132,13 +130,35 @@ export function* fetchLabels(action) {
...
+  const missingUnitIds = yield select(state => uids.filter(uid => !mdbSelectors.getUnitById(state.mdb, uid)));
+
+  if (missingUnitIds.length > 0) {
+    const language = yield select(state => settings.getLanguage(state.settings));
+    const { data } = yield call(Api.units, { id: missingUnitIds, language, pageSize: missingUnitIds.length });
+    yield put(actions.receiveContentUnits(data.content_units));
+  }
+}
+
+export function* fetchMissingCollections(uids) {
```
Comment (bbfsdev) [inline src/sagas/mdb.js:150]: same.

**Review by bbfsdev — CHANGES_REQUESTED** [review]
Demo: I was using this demo: http://bbdev6.kbb1.com:4506/he/programs

1. src/components/FiltersAside/helper.js is an empty file, remove it
2. Empty subcategories for programs and clips
image
3. Can you provide link where src/components/FiltersAside/CollectionFilter/CollectionItem.js used?
4. Why did you rewrite the programs page: src/components/Sections/Programs/tabs/Clips/List.js, src/components/Sections/Programs/MainPage.js, src/components/Sections/Programs/tabs/Programs/List.js

File: src/components/Sections/Topics/SubTopics.js (hunk 1 above)
Comment (davgur) [reply src/components/Sections/Topics/SubTopics.js:46]: on unmount component I clear filter input
image

File: src/sagas/recommended.js (hunk 6 above)
Comment (davgur) [reply src/sagas/recommended.js:271]: Want to make it public, so it will part on mdb and not recomindations

File: src/sagas/stats.js (hunk 5 above)
Comment (davgur) [reply src/sagas/stats.js:46]: stats now not part of list but part of filters. It have special actions

File: src/components/shared/TooltipIfNeed.js (hunk 4 above)
Comment (davgur) [reply src/components/shared/TooltipIfNeed.js:11]: I was add it cause props of this component are not clear (part of props are child props)

File: src/sagas/filtersAside.js [hunk 9]
```diff
@@ -38,26 +42,27 @@ export function* fetchStat(action) {
 
     const requests = [];
 
+    filterParams = { ...defaultStatParams, ...filterParams };
     requests.push(call(Api.unitsStats, filterParams));
-    requests.push(call(Api.collectionsStats, filterParams));
-    requests.push(call(Api.labelsStats, filterParams));
+    countC && requests.push(call(Api.collectionsStats, filterParams));
+    countL && requests.push(call(Api.labelsStats, filterParams));
     if (isFilteredByBase) {
```
Comment (bbfsdev) [inline src/sagas/filtersAside.js:49]: image

**Review by bbfsdev — CHANGES_REQUESTED** [review]

File: src/sagas/filtersAside.js (hunk 9 above)
Comment (davgur) [reply src/sagas/filtersAside.js:49]: I call it twice from component - one with isPrepare - true - with base params only. Its need for build tree of items - calls once. Second with filter params  - call every filter change

File: src/sagas/filtersAside.js [hunk 10]
```diff
@@ -1,20 +1,24 @@
...
+  with_sources: true,
+  with_tags: true,
+  with_collections: false,
+  with_languages: false,
+  with_content_types: true
 };
 
 export function* fetchStat(action) {
-  const { namespace, params, isPrepare } = action.payload;
+  const { namespace, params, options: { isPrepare, countC = false, countL = false } } = action.payload;
```
[the remark this answers is not in the document]
Comment (davgur) [reply src/sagas/filtersAside.js:21]: use it if need count not CU only but collections or labels too, dependence on page

File: src/sagas/mdb.js (hunk 7 above)
Comment (davgur) [reply src/sagas/mdb.js:140]: i plane use it but on this moment it was not need, you want I remove it from this PR?

File: src/components/Sections/Topics/TopicContainer.js (hunk 2 above)
Comment (davgur) [reply src/components/Sections/Topics/TopicContainer.js:110]: second param - base params - for example for topic page it will _tag=[tag uid]_

**Review by bbfsdev — APPROVED** [review]
